CRM/Upgrade/Incremental/php/FiveFortyThree.php

```diff
@@ -169,4 +170,19 @@ public static function fixRelationshipCacheTableCollation():bool {
     return TRUE;
   }
 
+  /**
+   * Ensure note_date is NOT NULL
+   *
+   * Since note_date is exposed in the UI, we want to ensure an empty value
+   * results in CURRENT_TIME, not a NULL date.
+   */
+  public static function setNoteDateFieldNotNull(): bool {
+    $i18nRewrite = FALSE;
+    $query = "ALTER TABLE civicrm_note CHANGE COLUMN `note_date` `note_date` timestamp NOT NULL DEFAULT current_timestamp() COMMENT 'Date attached to the note'";
+    CRM_Core_DAO::executeQuery($query, [], TRUE, NULL, FALSE, $i18nRewrite);
```

Comment:
The order of steps might be backwards here. If there are existing fields that contain null this should fail?

This could also be in the sql file instead of the php file - either way I think for this.

Reply:
You are right!! I just pushed a fix.